### PR TITLE
 On branch main-dev_readme_update

### PR DESCRIPTION
**Pull request by dwickelhaus**
Changes to be committed:
	modified:   README.md

  Added from V2.2 README.md file
    * `AdapterIdentity` - Adapter Identity name used to prefix dataitems within the Agent device ids and names. 
    *  #### MQTT Adapter/Source 
    * ### MQTT JSON Ingress Protocol Version 2.0 Corrected minor spelling errors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -948,6 +948,173 @@ Sinks {
...
+    *Required*
+
+* `MqttClientId` - Port number of MQTT Broker
+
+    *Default*: Auto-generated
+
+	> **⚠️Note:** Mqtt Sinks and Mqtt Adapters create separate connections to their respective brokers, but currently use the same client ID by default. Because of this, when using a single broker for source and sink, best practice is to explicitly specify their respective `MqttClientId`
+	>
+
+	> **⚠️Note:** Currently, there is no JSON parser functionality. Agent is expecting a raw SHDR-formatted string
```
Comment (robot-ranger) [inline README.md:984]: this is no longer accurate - there is injest protocol specs now (see lines 997+ below)

Comment (robot-ranger) [reply README.md:984]: (these were notes i had added before we developed the ingest protocol

Comment (dwickelhaus) [reply README.md:984]: I'll remove those on the next update.

Comment (dwickelhaus) [reply README.md:984]: Is the first note still relevant?

Comment (robot-ranger) [reply README.md:984]: > Is the first note still relevant?

yes, i believe so. iirc, the client id generation strategy is the same across all mqtt client instances which results in them all being the same client id. unless @wsobel has implemented something else.

**Comment by wsobel** [comment]
We have a json parser now. Regarding the second note. I have started a branch to do connection pooling for mqtt. BestW(Sent from mobile)On Jan 23, 2024, at 22:16, robot-ranger ***@***.***> wrote:﻿
@robot-ranger commented on this pull request.



In README.md:
> +* `MqttPort` - Port number of MQTT Broker
+
+    *Default*: 1883
+
+* `topics` - list of topics to subscribe to. Note : Only raw SHDR strings supported at this time
+
+    *Required*
+
+* `MqttClientId` - Port number of MQTT Broker
+
+    *Default*: Auto-generated
+
+	> **⚠️Note:** Mqtt Sinks and Mqtt Adapters create separate connections to their respective brokers, but currently use the same client ID by default. Because of this, when using a single broker for source and sink, best practice is to explicitly specify their respective `MqttClientId`
+	>
+
+	> **⚠️Note:** Currently, there is no JSON parser functionality. Agent is expecting a raw SHDR-formatted string


Is the first note still relevant?

yes, i believe so. iirc, the client id generation strategy is the same across all mqtt client instances which results in them all being the same client id. unless @wsobel has implemented something else.

**Comment by wsobel** [comment]
Looks good

**Comment by wsobel** [comment]
Forcing merge since readme does not trigger build.

**Pull request merged**